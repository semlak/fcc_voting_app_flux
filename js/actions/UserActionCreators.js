'use strict';

/*
UserActionCreators.js
 */

// import AppDispatcher from '../dispatcher/AppDispatcher';
// import UserConstants from '../constants/UserConstants';
import UserWebAPIUtils from '../utils/UserWebAPIUtils';

var UserActionCreators = {

	/**
	 * @param  {object} user, containing a username, password, and fullname.  (no 'role' provided, this will be set by server )
					The password will be hashed on the server. password will not be stored in plaintext after creation
	 */
	create: function(/*object*/ user) {
		// AppDispatcher.dispatch({
		//   actionType: UserConstants.USER_CREATE,
		//   username: user.username,
		//   password: user.password,
		//   fullname: user.fullname,
		//   role: user.role
		// });
		UserWebAPIUtils.registerNewUser(user);
	},

	/**
	 * @param  {string} id The ID of the User item
	 * @param  {object} userUpdates:  object containing possible updates (username, password, and/or fullname, role)
	 */
	update: function(id, /*object*/ userUpdates) {
		console.log('UserActionCreators.update function fired');

		// var data = userUpdates;
		// data.id = id;
		// AppDispatcher.dispatch({
		//   actionType: UserConstants.USER_UPDATE,
		//   id: id,
		//   userUpdates: userUpdates
		// });
		UserWebAPIUtils.update(id, userUpdates);
	},

	getAllFromServer: function() {
		// console.log('in UserActionCreators, getallFromServer. Dispatching USER_GET_ALL_FROM_SERVER')

		// AppDispatcher.dispatch({
		//   actionType: UserConstants.USER_GET_ALL_FROM_SERVER
		// });
		UserWebAPIUtils.getAllUsers();
	},
	/**
	 * Toggle whether a single User is complete
	 * @param  {object} user
	 */
	// toggleComplete: function(user) {
	//   var id = user.id;
	//   var actionType = user.complete ?
	//       UserConstants.USER_UNDO_COMPLETE :
	//       UserConstants.USER_COMPLETE;

	//   AppDispatcher.dispatch({
	//     actionType: actionType,
	//     id: id
	//   });
	// },

	/**
	 * Mark all Users as complete
	 */
	// toggleCompleteAll: function() {
	//   AppDispatcher.dispatch({
	//     actionType: UserConstants.USER_TOGGLE_COMPLETE_ALL
	//   });
	// },

	/**
	 * @param  {string} id
	 */
	// destroy: function(id) {
	// 	//note: currently, the server will not destroy any polls or votes associated with user when user is deleted
	// 	// UserWebAPIUtils.destroy(id)
	// 	// AppDispatcher.dispatch({
	// 	//   actionType: UserConstants.USER_DESTROY,
	// 	//   id: id
	// 	// });
	// },


	login: function(username, password) {
		UserWebAPIUtils.login(username, password);
		// AppDispatcher.dispatch({
		//   actionType: UserConstants.USER_LOGIN,
		//   username: username,
		//   password: password
		// })
	},

	logout: function() {
		UserWebAPIUtils.logout();
	}

	/**
	 * Delete all the completed Users
	 */
	// destroyCompleted: function() {
	//   AppDispatcher.dispatch({
	//     actionType: UserConstants.USER_DESTROY_COMPLETED
	//   });
	// }

};

module.exports = UserActionCreators;
