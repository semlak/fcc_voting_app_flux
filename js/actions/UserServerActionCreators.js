'use strict';


import AppDispatcher from '../dispatcher/AppDispatcher';
import UserConstants from '../constants/UserConstants';

module.exports = {

	receiveAll: function(rawUsers) {
		// console.log('in UserServerActionCreators, received the receiveAll signal. dispatching the USER_RECEIVE_RAW_USERS signal, rawUsers are ', rawUsers)
		AppDispatcher.dispatch({
			actionType: UserConstants.USER_RECEIVE_RAW_USERS,
			rawUsers: rawUsers
		});
	},

	receiveCreatedUser: function(createdUser, message) {
		//errorMessage will be null if user was successfully created.
		// console.log("in UserServerActionCreators, receiveCreatedUser; createdUser is ", createdUser);
		AppDispatcher.dispatch({
			actionType: UserConstants.USER_CREATE,
			rawUser: createdUser,
			errorMessage: createdUser ==  null ? message : null,
			successMessage: createdUser !=  null ? message : null
		});
		// if (createdUser != null) {
		// 	// console.log("in if branch of receiveCreatedUser");
		// 	// console.log("createdUser is ",  createdUser)
		// 	AppDispatcher.dispatch({
		// 		actionType: UserConstants.USER_CREATE,
		// 		rawUser: createdUser,
		// 		errorMessage: null,
		// 		successMessage: message
		// 	});
		// }
		// else {
		// 	// console.log("in else branch of receiveCreatedUser");
		// 	AppDispatcher.dispatch({
		// 		actionType: UserConstants.USER_CREATE,
		// 		rawUser: null,
		// 		errorMessage: message,
		// 		successMessage: null
		// 	});
		// }
	},

	receiveUpdatedUser: function(updatedUser, message_obj) {
		//errorMessage will be null if user was successfully created.
		// console.log("in UserServerActionCreators, receiveUpdatedUser; updatedUser is ", updatedUser);
		AppDispatcher.dispatch({
			actionType: UserConstants.USER_UPDATE,
			rawUser: updatedUser,
			message_obj: message_obj
		});
	},

	setAuthenticatedUserState: function(rawUser, message_obj) {
		AppDispatcher.dispatch({
			actionType: UserConstants.USER_SET_AUTHENTICATED_USER_STATE,
			rawUser: rawUser,
			message_obj: message_obj
		});
	}

};
