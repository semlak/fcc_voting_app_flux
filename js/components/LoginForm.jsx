import React from 'react'
import UserStore from '../stores/UserStore';
import UserActionCreators from '../actions/UserActionCreators';
import {Button, Grid, Row, Col, Form, FormGroup, FormControl, Checkbox, ControlLabel, HelpBlock} from 'react-bootstrap'
import ReactPropTypes from 'react/lib/ReactPropTypes';


export default React.createClass({
	getInitialState: function() {
		return {
				username: '',
				password: '',
				message_obj: null
		};
	},

  componentDidMount: function() {
    UserStore.addAuthenticationChangeListener(this._onAuthenticationChange);
  },

  componentWillUnmount: function() {
    UserStore.removeAuthenticationChangeListener(this._onAuthenticationChange);
  },

	handleLogin: function() {
		UserActionCreators.login(this.state.username, this.state.password)
	},

	handleFieldChange: function(e) {
		// console.log(e.target.name, e.target.value)
		var key = e.target.name.toString();
		var newParam = {}
		newParam[key] = e.target.value
		newParam['message_obj'] = null;
		this.setState(newParam)
	},
	render: function() {
		// console.log("rendering LoginForm")
		var message_obj = this.state.message_obj;
		var validationState = message_obj == null ? null : (message_obj.error ? "error" : "success");
		var validationMessage = message_obj == null ? "" : message_obj.message_text;
		return (
			<div className="sign_in_form" onClick={this.handleClick} >
				<div className="modal-body" onClick={this.handleClick} >
				  <Form horizontal>
				  	<FormGroup validationState={validationState}>
					    <FormGroup controlId="formHorizontalUsername">
					      <Col smOffset={0} sm={2} componentClass={ControlLabel}>
					        Username:
					      </Col>
					      <Col sm={10}>
					        <FormControl type="text" name="username" value={this.state.username} onChange={this.handleFieldChange} placeholder='Enter Username' />
					      </Col>
					    </FormGroup>
					    <FormGroup controlId="formHorizontalPassword">
					      <Col smOffset={0} sm={2} componentClass={ControlLabel}>
					        Password:
					      </Col>
					      <Col sm={10}>
					        <FormControl type="password" name="password" value={this.state.password} onChange={this.handleFieldChange} placeholder='Enter Password'/>
					      </Col>
					    </FormGroup>
				    	<Col smOffset={2} sm={10}>
								{validationState == null ? <HelpBlock>{"Log in with your username and password."}</HelpBlock> : <HelpBlock>{validationMessage}</HelpBlock> }
							</Col>
					  </FormGroup>
				  </Form>
				</div>
				<div className="modal-footer">
					<Button bsStyle="primary" onClick={this.handleLogin}>Sign In</Button>
					<Button bsStyle='default' onClick={this.props.onCancel}>Cancel </Button>
				</div>
			</div>
		);
	},

  _onAuthenticationChange: function(message_obj) {
    // this.setState(getUserState());
    if (message_obj != null && message_obj.error) {
    	this.setState({message_obj: message_obj});
    }
  }
})