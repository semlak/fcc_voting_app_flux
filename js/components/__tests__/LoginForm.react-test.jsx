'use strict';

jest.autoMockOff();

import LoginForm from '../LoginForm';
import React from 'react';
// import Link from '../Link.react';
import renderer from 'react-test-renderer';
// import {searchTree, searchTreeForProps, searchTreeForClassName} from '../../testing/extraFunctions';
import {searchTreeForProps} from '../../testing/extraFunctions';

jest.mock('react-dom');
jest.mock('../../utils/UserWebAPIUtils.js');
import UserActionCreators from '../../actions/UserActionCreators';
import UserServerActionCreators from '../../actions/UserServerActionCreators';

var actualHandleLoginFunction = function(username, password) {
		UserActionCreators.login(username, password);
};

describe('LoginForm', function() {
	// var LoginForm;
	var onCancel;
	var handleLogin;

	beforeEach(function() {
		// LoginForm = require('../LoginForm');
		// console.log('\n\n\Pll is', Poll)
		// done()
		onCancel = jest.fn();
		handleLogin = jest.genMockFunction();

	});

	it('has has username and password input fields', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
		// console.log('tree.children:', tree.children);
		expect(tree.children.length).toBe(2);

		expect(tree.children[0].props.className).toBe('modal-body');

		let usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		let passwordNode = searchTreeForProps(tree, {name: 'password'})[0];
		expect(usernameNode.props.name).toBe('username');
		expect(usernameNode.props.value).toBe('');
		expect(usernameNode.props.placeholder).toBe('Enter Username');
		expect(usernameNode.props.type).toBe('text');
		expect(usernameNode.props.className).toBe('form-control');


		expect(passwordNode.props.name).toBe('password');
		expect(passwordNode.props.value).toBe('');
		expect(passwordNode.props.placeholder).toBe('Enter Password');
		expect(passwordNode.props.type).toBe('password');
		expect(passwordNode.props.className).toBe('form-control');


	});


	it('responds to changes in username and password fields', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		// expect(tree).toMatchSnapshot();
		// console.log('tree.children:', tree.children);

		let usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		let passwordNode = searchTreeForProps(tree, {name: 'password'})[0];

		let e1 = {
			target: {
				name: 'username',
				value: 'simple_username84'
			}
		};
		usernameNode.props.onChange(e1);

		tree = component.toJSON();
		usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		expect(usernameNode.props.value).toBe('simple_username84');
		// console.log('usernameNode:', usernameNode);

		let e2 = {
			target: {
				name: 'password',
				value: 'dumb_pa$$word'
			}
		}

		passwordNode.props.onChange(e2);
		tree = component.toJSON();
		passwordNode = searchTreeForProps(tree, {name: 'password'})[0];
		expect(passwordNode.props.value).toBe('dumb_pa$$word');

		expect(tree).toMatchSnapshot();

	});



	it('calls the handleLogin callback with username and password as arguments when sign-in button clicked', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		// expect(tree).toMatchSnapshot();

		let usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		let passwordNode = searchTreeForProps(tree, {name: 'password'})[0];

		let e1 = {
			target: {
				name: 'username',
				value: 'simple_username84'
			}
		};

		usernameNode.props != null && usernameNode.props.onChange(e1);


		let e2 = {
			target: {
				name: 'password',
				value: 'dumb_pa$$word',
			}
		}

		passwordNode.props != null && passwordNode.props.onChange(e2);


		// tree = component.toJSON();
		// usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		// expect(usernameNode.props.value).toBe('simple_username84');
		// console.log('usernameNode:', usernameNode);

		// tree = component.toJSON();
		// passwordNode = searchTreeForProps(tree, {name: 'password'})[0];
		// expect(passwordNode.props.value).toBe('dumb_pa$$word');
		let loginButton = searchTreeForProps(tree, {id: 'login-button'})[0];
		// console.log('loginButton:', loginButton);
		loginButton.props != null && loginButton.props.onClick();

		expect(handleLogin).toHaveBeenCalledTimes(1);
		expect(handleLogin).toBeCalledWith('simple_username84', 'dumb_pa$$word');
		expect(handleLogin).lastCalledWith('simple_username84', 'dumb_pa$$word');

	});






	it('automatically calls the handleLogin callback if \'enter\' key is pressed while either input field is active', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		// expect(tree).toMatchSnapshot();

		let usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		let passwordNode = searchTreeForProps(tree, {name: 'password'})[0];

		let e1 = {
			target: {
				name: 'username',
				value: 'simple_username84'
			}
		};
		usernameNode.props != null && usernameNode.props.onChange(e1);

		tree = component.toJSON();
		usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		expect(usernameNode.props.value).toBe('simple_username84');
		// console.log('usernameNode:', usernameNode);

		let e2 = {
			target: {
				name: 'password',
				value: 'dumb_pa$$word',
			}
		}

		let e3 = {
			target: {
				name: 'password',
				value: 'dumb_pa$$word',
			},
			key: 'Enter'
		}

		passwordNode.props != null && passwordNode.props.onChange(e2);
		tree = component.toJSON();
		passwordNode = searchTreeForProps(tree, {name: 'password'})[0];
		expect(passwordNode.props.value).toBe('dumb_pa$$word');

		let node3 = searchTreeForProps(tree, {className: 'sign-in-form'})[0];
		node3.props != null && node3.props.onKeyPress(e3);
		expect(handleLogin).toHaveBeenCalledTimes(1);


	});


	it('calls the onCancel callback when cancel button is clicked', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		// expect(tree).toMatchSnapshot();

		let cancelButton = searchTreeForProps(tree, {id: 'cancel-button'})[0];
		// console.log('loginButton:', loginButton);
		cancelButton.props != null && cancelButton.props.onClick();

		expect(onCancel).toHaveBeenCalledTimes(1);

	});




	it('displays error message to user when no username is provided on sign-in attempt, and whatever password entered remains', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		// expect(tree).toMatchSnapshot();

		let passwordNode = searchTreeForProps(tree, {name: 'password'})[0];

		let e2 = {
			target: {
				name: 'password',
				value: 'dumb_pa$$word'
			}
		};

		passwordNode.props != null && passwordNode.props.onChange(e2);

		let loginButton = searchTreeForProps(tree, {id: 'login-button'})[0];
		loginButton.props != null && loginButton.props.onClick();

		tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		// Because the LoginForm has its own function called onClick, the handleLogin mockFunction callback is only called if the username and password are not empty
		expect(handleLogin).toHaveBeenCalledTimes(0);
		expect(handleLogin).not.toHaveBeenCalled();

		let helpBlock = searchTreeForProps(tree, {className: 'help-block'})[0];
		// console.log('helpBlock:', helpBlock);
		expect(helpBlock.children[0]).toBe('Username must be at least one character long.');

		passwordNode = searchTreeForProps(tree, {name: 'password'})[0];
		expect(passwordNode.props.value).toBe('dumb_pa$$word');

		// let formGroups = searchTreeForProps(tree, {className: 'form-group'});
		let topFormGroup = searchTreeForProps(tree, {id: 'top-form-group'})[0];
		// console.log('formGroups:', formGroups);
		// console.log('topFormGroup:', topFormGroup);
		expect(topFormGroup.props.className).toMatch('has-error');

	});



	it('displays error message to user when no password is provided on sign-in attempt, and whatever username entered remains', () => {

		var component = renderer.create(
			<LoginForm
				onCancel={onCancel}
				handleLogin={handleLogin}
			/>
		);

		let tree = component.toJSON();
		// expect(tree).toMatchSnapshot();

		let usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		let passwordNode = searchTreeForProps(tree, {name: 'password'})[0];

		let e1 = {
			target: {
				name: 'username',
				value: 'simple_username84'
			}
		};

		usernameNode.props != null && usernameNode.props.onChange(e1);

		let loginButton = searchTreeForProps(tree, {id: 'login-button'})[0];
		loginButton.props != null && loginButton.props.onClick();

		tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		// Because the LoginForm has its own function called onClick, the handleLogin mockFunction callback is only called if the username and password are not empty
		expect(handleLogin).toHaveBeenCalledTimes(0);
		expect(handleLogin).not.toHaveBeenCalled();

		usernameNode = searchTreeForProps(tree, {name: 'username'})[0];
		expect(usernameNode.props.value).toBe('simple_username84');

		let helpBlock = searchTreeForProps(tree, {className: 'help-block'})[0];
		// console.log('helpBlock:', helpBlock);
		expect(helpBlock.children[0]).toBe('Password must be at least one character long.');


		let topFormGroup = searchTreeForProps(tree, {id: 'top-form-group'})[0];
		expect(topFormGroup.props.className).toMatch('has-error');

	});


});
