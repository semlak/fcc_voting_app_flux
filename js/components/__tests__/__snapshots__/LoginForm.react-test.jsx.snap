exports[`LoginForm displays error message to user when no password is provided on sign-in attempt, and whatever username entered remains 1`] = `
<div
  className="sign-in-form"
  onKeyPress={[Function bound onKeyPress]}>
  <div
    className="modal-body">
    <form
      className="form-horizontal">
      <div
        className="form-group has-error"
        id="top-form-group">
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalUsername">
            Username:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalUsername"
              name="username"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Username"
              type="text"
              value="simple_username84" />
          </div>
        </div>
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalPassword">
            Password:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalPassword"
              name="password"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Password"
              type="password"
              value="" />
          </div>
        </div>
        <div
          className="col-sm-10 col-sm-offset-2">
          <span
            className="help-block">
            Password must be at least one character long.
          </span>
        </div>
      </div>
    </form>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-primary"
      disabled={false}
      id="login-button"
      onClick={[Function bound handleLogin]}
      type="button">
      Sign In
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      id="cancel-button"
      onClick={[Function mockConstructor]}
      type="button">
      Cancel
    </button>
  </div>
</div>
`;

exports[`LoginForm displays error message to user when no username is provided on sign-in attempt, and whatever password entered remains 1`] = `
<div
  className="sign-in-form"
  onKeyPress={[Function bound onKeyPress]}>
  <div
    className="modal-body">
    <form
      className="form-horizontal">
      <div
        className="form-group has-error"
        id="top-form-group">
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalUsername">
            Username:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalUsername"
              name="username"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Username"
              type="text"
              value="" />
          </div>
        </div>
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalPassword">
            Password:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalPassword"
              name="password"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Password"
              type="password"
              value="dumb_pa$$word" />
          </div>
        </div>
        <div
          className="col-sm-10 col-sm-offset-2">
          <span
            className="help-block">
            Username must be at least one character long.
          </span>
        </div>
      </div>
    </form>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-primary"
      disabled={false}
      id="login-button"
      onClick={[Function bound handleLogin]}
      type="button">
      Sign In
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      id="cancel-button"
      onClick={[Function mockConstructor]}
      type="button">
      Cancel
    </button>
  </div>
</div>
`;

exports[`LoginForm has has username and password input fields 1`] = `
<div
  className="sign-in-form"
  onKeyPress={[Function bound onKeyPress]}>
  <div
    className="modal-body">
    <form
      className="form-horizontal">
      <div
        className="form-group"
        id="top-form-group">
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalUsername">
            Username:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalUsername"
              name="username"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Username"
              type="text"
              value="" />
          </div>
        </div>
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalPassword">
            Password:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalPassword"
              name="password"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Password"
              type="password"
              value="" />
          </div>
        </div>
        <div
          className="col-sm-10 col-sm-offset-2">
          <span
            className="help-block">
            Log in with your username and password.
          </span>
        </div>
      </div>
    </form>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-primary"
      disabled={false}
      id="login-button"
      onClick={[Function bound handleLogin]}
      type="button">
      Sign In
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      id="cancel-button"
      onClick={[Function mockConstructor]}
      type="button">
      Cancel
    </button>
  </div>
</div>
`;

exports[`LoginForm responds to changes in username and password fields 1`] = `
<div
  className="sign-in-form"
  onKeyPress={[Function bound onKeyPress]}>
  <div
    className="modal-body">
    <form
      className="form-horizontal">
      <div
        className="form-group"
        id="top-form-group">
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalUsername">
            Username:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalUsername"
              name="username"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Username"
              type="text"
              value="simple_username84" />
          </div>
        </div>
        <div
          className="form-group">
          <label
            className="col-sm-2 col-sm-offset-0 control-label"
            htmlFor="formHorizontalPassword">
            Password:
          </label>
          <div
            className="col-sm-10">
            <input
              className="form-control"
              id="formHorizontalPassword"
              name="password"
              onChange={[Function bound handleFieldChange]}
              placeholder="Enter Password"
              type="password"
              value="dumb_pa$$word" />
          </div>
        </div>
        <div
          className="col-sm-10 col-sm-offset-2">
          <span
            className="help-block">
            Log in with your username and password.
          </span>
        </div>
      </div>
    </form>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-primary"
      disabled={false}
      id="login-button"
      onClick={[Function bound handleLogin]}
      type="button">
      Sign In
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      id="cancel-button"
      onClick={[Function mockConstructor]}
      type="button">
      Cancel
    </button>
  </div>
</div>
`;
